gobblin-service/src/main/java/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -0,0 +1,293 @@
+ * and runs them via {@link Orchestrator}.
+ */
+@Alpha
+public class GobblinServiceJobScheduler extends JobScheduler implements SpecCatalogListener {
+  protected final Logger _log;
+
+  protected final Optional<FlowCatalog> flowCatalog;
+  protected final Optional<HelixManager> helixManager;
+  protected final Orchestrator orchestrator;
+  protected final Map<String, Spec> scheduledFlowSpecs;
+
+  @Getter
+  protected volatile boolean isActive;
+
+  public GobblinServiceJobScheduler(Config config, Optional<HelixManager> helixManager, Optional<FlowCatalog> flowCatalog,
+      Optional<TopologyCatalog> topologyCatalog, Orchestrator orchestrator, SchedulerService schedulerService,
+      Optional<Logger> log) throws Exception {
+    super(ConfigUtils.configToProperties(config), schedulerService);
+
+    _log = log.isPresent() ? log.get() : LoggerFactory.getLogger(getClass());
+
+    this.flowCatalog = flowCatalog;
+    this.helixManager = helixManager;
+    this.orchestrator = orchestrator;
+    this.scheduledFlowSpecs = Maps.newHashMap();
+  }
+
+  public GobblinServiceJobScheduler(Config config, Optional<HelixManager> helixManager, Optional<FlowCatalog> flowCatalog,
+      Optional<TopologyCatalog> topologyCatalog, SchedulerService schedulerService, Optional<Logger> log) throws Exception {
+    this(config, helixManager, flowCatalog, topologyCatalog, new Orchestrator(config, topologyCatalog, log),
+        schedulerService, log);
+  }
+
+
+  public synchronized void setActive(boolean isActive) {
+    if (this.isActive == isActive) {
+      // No-op if already in correct state
+      return;
+    }
+    this.isActive = isActive;
+
+    if (isActive) {
+      if (this.flowCatalog.isPresent()) {
+        Collection<Spec> specs = this.flowCatalog.get().getSpecs();
+        for (Spec spec : specs) {
+          onAddSpec(spec);
+        }
+      }
+    } else {
+      for (Spec spec : this.scheduledFlowSpecs.values()) {
+        onDeleteSpec(spec.getUri(), spec.getVersion());
+      }
+    }
+  }
+
+  @Override
+  protected void startUp() throws Exception {
+    super.startUp();
+  }
+
+  @Override
+  public void scheduleJob(Properties jobProps, JobListener jobListener) throws JobException {
+    Map<String, Object> additionalJobDataMap = Maps.newHashMap();
+    additionalJobDataMap.put(ServiceConfigKeys.GOBBLIN_SERVICE_FLOWSPEC,
+        this.scheduledFlowSpecs.get(jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY)));
+
+    try {
+      scheduleJob(jobProps, jobListener, additionalJobDataMap, GobblinServiceJob.class);
+    } catch (Exception e) {
+      throw new JobException("Failed to schedule job " + jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY), e);
+    }
+  }
+
+  @Override
+  public void runJob(Properties jobProps, JobListener jobListener) throws JobException {
+    try {
+      Spec flowSpec = this.scheduledFlowSpecs.get(jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY));
```

Comment:
Removed schedule while compiling in compiler, because I think its more of a property of compiler to decide if the schedule should be passed along in the JobSpec or not. Any contrary thoughts?